spec/mem/pointer.md

```diff
@@ -106,11 +106,46 @@ pub enum LayoutStrategy {
     },
 }
 
+/// Corresponds to the variants of `LayoutStrategy`
+pub enum UnsafeCellStrategy {
+    /// List of [start, end) ranges.
+    Sized { bytes: List<(Offset, Offset)> },
+    /// Since the elements of a slice have the same type, we only keep track of the
+    /// UnsafeCell's of one element, we can then "repeat" this for the rest of the slice.
+    Slice { element: List<(Offset, Offset)> },
+    TraitObject { is_freeze: bool },
+    Tuple {
+        head: List<(Offset, Offset)>,
+        #[specr::indirection]
+        tail: UnsafeCellStrategy,
+    },
+}
+
+impl UnsafeCellStrategy {
+    pub fn is_freeze(self) -> bool {
```

Comment:
This method should have a comment explaining that it's used to figure out whether the "outside" range of a retag is freeze.